test/beacon/Voucher.test.ts

```diff
@@ -405,5 +409,151 @@ describe('Voucher', function () {
                 ),
             ).to.be.revertedWith('IexecPocoMock: Failed to sponsorMatchOrders');
         });
+        describe('Match orders boost', async function () {
+            it('Should match orders boost with full sponsored amount', async () => {
+                for (const asset of [app, dataset, workerpool]) {
+                    await voucherHubWithAssetEligibilityManagerSigner
+                        .addEligibleAsset(voucherType, asset)
+                        .then((x) => x.wait());
+                }
+                const voucherInitialCreditBalance = await voucher.getBalance();
+                const voucherInitialSrlcBalance = await getVoucherBalanceOnIexecPoco();
+                const requesterInitialSrlcBalance = await getRequesterBalanceOnIexecPoco();
+
+                expect(
+                    await voucherWithOwnerSigner.matchOrdersBoost.staticCall(
+                        appOrder,
+                        datasetOrder,
+                        workerpoolOrder,
+                        requestOrder,
+                    ),
+                ).to.be.equal(dealId);
+                await expect(
+                    voucherWithOwnerSigner.matchOrdersBoost(
+                        appOrder,
+                        datasetOrder,
+                        workerpoolOrder,
+                        requestOrder,
+                    ),
+                )
+                    .to.emit(voucher, 'OrdersBoostMatchedWithVoucher')
+                    .withArgs(dealId);
+                expect(await voucher.getBalance())
+                    .to.be.equal(voucherInitialCreditBalance - dealPrice)
+                    .to.be.equal(await getVoucherBalanceOnIexecPoco())
+                    .to.be.equal(voucherInitialSrlcBalance - dealPrice);
+                expect(await getRequesterBalanceOnIexecPoco()).to.be.equal(
+                    requesterInitialSrlcBalance,
+                );
+                expect(await voucher.getSponsoredAmount(dealId)).to.equal(dealPrice);
+            });
+
+            it('Should match orders boost with full non-sponsored amount', async () => {
```

Comment:
Can we add a test for partial sponsoring?

Reply:
Done [here](https://github.com/iExecBlockchainComputing/iexec-voucher-contracts/pull/16/commits/bb6bda7d7c7c128ad0c9e95a8a13098a3981f0aa) let me know if that's ok for you